pkg/coveraggregator/simple_file_aggregator.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package coveraggregator
+
+func MakeSimpleFileLineCoverAggregator(
+	fileVersions map[RepoBranchCommit]string, base RepoBranchCommit) FileCoverageAggregator {
+	baseFile := ""
+	baseFileExists := false
+	for rbc, fileContent := range fileVersions {
+		if rbc == base {
+			baseFile = fileContent
+			baseFileExists = true
+			break
+		}
+	}
+	if !baseFileExists {
+		return &DeletedFileLineAggregator{}
+	}
+	a := &SimpleFileLineCoverAggregator{
+		rbcToFile: fileVersions,
+		baseFile:  baseFile,
+		hitCounts: make(map[int]int),
+		matchers:  make(map[RepoBranchCommit]*LineToLineMatcher),
+	}
+	for rbc, fileVersion := range fileVersions {
+		a.matchers[rbc] = makeLineToLineMatcher(fileVersion, baseFile)
+	}
+	return a
+}
+
+type SimpleFileLineCoverAggregator struct {
```

Comment:
Will we have another non-simple aggregator next to this one? If not, remove "simple", the _the_ aggregator we have.

Reply:
Yes, we'll have. At least I'd like to.
For gVisor we can do more interesting aggregations.
And having the basic blocks information we can also calculate coverage differently.